nio/events/ephemeral.py

```diff
@@ -84,3 +86,50 @@ class TypingNoticeEvent(EphemeralEvent):
     @verify_or_none(Schemas.m_typing)
     def from_dict(cls, parsed_dict):
         return cls(parsed_dict["content"]["user_ids"])
+
+
+@attr.s
+class Receipt(object):
+    """Receipt of a user acknowledging an event.
+    
+    If `receipt_type` is "m.read", then it is a read receipt and shows the last
+    event that a user has read.
+
+    Attributes:
+        event_id (str): the ID of the event being acknowleged
+        receipt_type (str): the type of receipt being received; this is
+            commonly "m.read" for read receipts.
+        user_id (str): the ID of the user who is acknowledging the event.
+        timestamp (int): The timestamp the receipt was sent at.
+    """
+    event_id = attr.ib(type=str)
+    receipt_type = attr.ib(type=str)
+    user_id = attr.ib(type=str)
+    timestamp = attr.ib(type=int)
+
+@attr.s
+class ReceiptEvent(EphemeralEvent):
+    """Informs the client of changes in the newest events seen by users.
+
+    A ReceiptEvent can contain multiple event_ids seen by many different users.
+    At the time of writing, all Receipts have a `receipt_type` of "m.read" and
+    are read receipts, but this may change in the future.
+
+    Attributes:
+        receipts (List[Receipt]): The list of `Receipt`s in this event.
+    """
+    receipts = attr.ib(type=List[Receipt])
+
+    @classmethod
+    @verify_or_none(Schemas.m_receipt)
+    def from_dict(cls, parsed_dict):
+        event_receipts: List[Receipt] = []
+
+        for event_id, event in parsed_dict["content"].items():
+            for receipt_type, receipt in event.items():
+                for user_id, user in receipt.items():
+                    event_receipts.append(
+                        Receipt(event_id, receipt_type, user_id, user["ts"])
```

Comment:
The ts isn't listed as required in the json schema. We either need to make it required or use `.get()` to fetch a None if it's missing.

Reply:
I've never used schema like this before, but I think the commit I pushed is what we want? I figured it's the only required thing since if the other fields are missing (event and user IDs) it's not an issue, since we iterate over the list of items that _do_ exist.